sysml/src/examples/Simple Tests/EnumerationTest.sysml

```diff
@@ -0,0 +1,35 @@
+package EnumerationTest {
+	
+	attribute def Color {
+		attribute val : ScalarValues::Natural;
+	}
+	
+	/**
+	 * An EnumerationDefinition can contain only EnumerationUsages. However,
+	 * it can specialize an AttributeDefinition in order to inherit
+	 * common features for its enumeration values.
+	 */
+	enum def ColorKind :> Color {
+		enum red {
+			:>> val = 0;
+		}
+		enum blue {
+			:>> val = 1;
+		}
+		enum green {
+			:>> val = 2;
+		}
+	}
+	
+	enum color : ColorKind;
```

Comment:
For me, allowing this use of the `enum` keyword as an alternative for an enum attribute is potentially confusing, because in the future it would not be clear whether the usage is an attribute, an item, a part, ... Also it is effectively a homonym with the `enum` used for the 'enumeration literals'. I would propose to reserve the sole `enum` keyword only for declaration of those 'enumeration literals'.

I understand that with `enum def` the logical consequence is to also have `enum` as a usage. Perhaps we could instead use:
```
    enum attribute def ColorKind :> Color {
        enum red {...}
    ...
    }
```
as a kind of special / qualified attribute and then only allow usage of such an attribute type via the normal `attribute` keyword, like in the next line `attribute color1 : ColorKind = ColorKind::blue;`. The same pattern could later also be used for Occurrences and Objects, i.e. `enum item def`, `enum part def`, `enum port def`. Their related `item`, `part` and `port` usages would then use existing normal syntax.

Reply:
The enumerated values of an enumeration definition _are_ enumeration usages, syntactically the same thing as enumeration usages used outside of an enumeration definition. So, the use of the `enum` keyword isn't really a "homonym" as such, but just a consistent use of that keyword to mark enumeration usages.

On the wider point of ultimately extending the syntax to allow enumerations of kinds of things other than attributes, I also felt some concern about that. But I am also loath to add complexity to the syntax at this point when we don't know for sure what the general capability will be yet, or even if we will have time to get it into the language. And, even if we ultimately have general syntax like `enum attribute def`, `enum part def`, etc., I think I would want to keep `enum def` as a shorthand for `enum attribute def`, because that is still going to be very common and should be simple.

On the other hand, I would agree that allowing enumeration usages outside enumeration definitions doesn't really provide any additional capability over just using an attribute usage typed by an enumeration definition. Currently, however, the abstract syntax allows the use of enumeration usages like any other usages, and if we allow that in the abstract syntax, we should have a way to notate it in the concrete syntax. But perhaps we shouldn't allow that in the abstract syntax, either -- it would be easy to require that an enumeration usage be owned by an enumeration definition.

It might be worth a discussion with Track 2 (or Sandy, at least) whether it would be more confusing at this point to allow enumeration usages outside of enumeration definitions for now, when that might not make sense in the future, or to introduce `enum def`s as the syntax now, but without allowing corresponding `enum`s outside of the defs.